hack/docker.sh

```diff
@@ -38,6 +39,11 @@ docker-login() {
             gcloud auth configure-docker --quiet 2> /dev/null
         ;;
 
+        docker.io/*)
```

Comment:
Not from your PR but should we not use a similar pattern with `docker.elastic.co` ? Otherwise any image with `/eck/` in its name matches the first condition, even if we actually want to push it to GCR or DockerHub.

I was thinking about something along those lines:

```bash
shopt -s extglob
...
    case "$image" in

        docker.elastic.co/@(eck|eck-ci|eck-snapshots)/*)
            echo "Authentication to ${registry}..."
            docker login -u "${DOCKER_LOGIN}" -p "${DOCKER_PASSWORD}" docker.elastic.co 2> /dev/null
        ;;

        *.gcr.io/*)
            echo "Authentication to ${registry}..."
            gcloud auth configure-docker --quiet 2> /dev/null
        ;;

        docker.io/*)
            echo "Authentication to ${registry}..."
            docker login -u "${DOCKERHUB_LOGIN}" -p "${DOCKERHUB_PASSWORD}" 2> /dev/null
        ;;

        *)
            if ! grep -q "$registry" ~/.docker/config.json; then
               echo "Please log in to $registry."
               exit 1
            fi
        ;;
    esac
```
_(not sure that `@(eck|eck-ci|eck-snapshots)` is really required)_

Reply:
+1 to use `docker.elastic.co/*)`.

Reply:
makes a lot of sense.  👍 updated.